Symantec_docs_code.md

```diff
@@ -0,0 +1,112 @@
+
+Doc management was continually concerned that putting a writer into the code -- writing code comments and code samples -- would set developer expectations of the writers too high. In other words, it was seen as an un-extensible model. But the web services project was pretty well contained (I continued to work on other aspects of the server docs), and in fact there was quite a lot of code savvy on our docs team as a whole that management had previously been unaware of.
+
+The biggest difficulty came when we moved from SOAP to REST. Code comments don't work as source content for REST web services documentation, and it took us a while to figure out what to use instead. The tooling that we eventually settled on, spring-restdocs, required even tighter integration of dev and doc effort because the main doc files are written by hand, but the example requests and responses are generated from a documentation method that's added to the mock tests. So dev has to write the test method, either dev or the writer creates the main source files that include the examples, and then any reference information must be maintained manually outside the code, but within the code framework. These issues were still being addressed when I left the company.
+
+Generally the kind of approach that we took is at least initially writer-intensive, and therefore not extensible without careful planning and attention to the nuances of load balancing. For the first set of web services that we shipped, I worked on the project full-time. For updates the following year, we had enough of a system in place that I could work on the updates about half time. But juggling a developer workflow and a separate documentation workflow was quite a challenge, especially when I came back to the developer workflow after nearly a year of work in the separate doc flow. Developer tools and configurations had changed, and I had to learn them all over again. I don't necessarily recommend an all-or-nothing approach (all developer tools all the time or else none), but it's important for everyone -- doc teams and dev teams -- to understand the difficulties of jumping into and out of different workflows and toolchains. 
+
+## What you wish you knew
+
+The biggest thing we didn't do was integrate the Javadoc build completely into the server build. Doing so would have required an investment of time and effort up front plus ongoing time and work to build the server locally, so we had good reasons for making the choice that we did. That said, automated builds would not only have saved writer time and effort, they would also have saved the occasional server build break (when the built docs caused the break, because they weren't part of the server tests), and they would also have guaranteed up-to-date Javadocs.
+
+On the other hand, because other doc components besides the Javadoc could only be built manually, more automation could have solved only part of the problem of keeping content up-to-date on the server.
+
```

Comment:
Was it a dev resources problem where doc builds were not prioritized high enough as compared to work on the services? Maybe say more about root causes for the decision not to automate... as in, management didn't want to invest, or I (as the writer) didn't find a way to do it myself? This is a great piece. Or maybe it was that the automation wasn't quite big enough for attention (that's sort of what I get from the second paragraph). What would you suggest to others who struggle with something similar (how big is big enough, what prevented automation)?